apps/analysis/serializers.py

```diff
@@ -17,65 +18,84 @@
 )
 
 
-class AnalysisPillarSerializer(UserResourceSerializer):
-    assignee_name = serializers.CharField(source='assignee.username', read_only=True)
-    analysis_title = serializers.CharField(source='analysis.title', read_only=True)
+class AnlyticalEntriesSerializer(UserResourceSerializer):
+    class Meta:
+        model = AnalyticalStatementEntry
+        fields = ('id', 'order', 'entry')
+        read_only_fields = ('analytical_statement',)
+
+
+class AnalyticalStatementSerializer(
+    RemoveNullFieldsMixin,
+    DynamicFieldsMixin,
+    UserResourceSerializer,
+    NestedCreateMixin,
+    NestedUpdateMixin,
+):
+    analytical_entries = AnlyticalEntriesSerializer(source='analyticalstatemententry_set', many=True, required=False)
 
     class Meta:
-        model = AnalysisPillar
+        model = AnalyticalStatement
         fields = '__all__'
-        read_only_fields = ('analysis',)
+        read_only_fields = ('analysis_pillar',)
 
     def validate(self, data):
-        analysis_id = self.context['view'].kwargs.get('analysis_id', None)
-        if analysis_id:
-            data['analysis_id'] = int(analysis_id)
+        analysis_pillar_id = self.context['view'].kwargs.get('analysis_pillar_id', None)
+        if analysis_pillar_id:
+            data['analysis_pillar_id'] = int(analysis_pillar_id)
+        # Validate the analytical_entries
+        entries = data.get('analyticalstatemententry_set')
+        if entries and len(entries) >= settings.ANALYTICAL_ENTRIES_COUNT:
+            raise serializers.ValidationError(f'Analytical entires count must be less than\
+                                              {settings.ANALYTICAL_ENTRIES_COUNT}')
```

Comment:
Let's do this.
- We don't want count to be > 50 not >= 50.
- If we don't have to break a string, let's not break, just move it to a new line.

```suggestion
        if entries and len(entries) > settings.ANALYTICAL_ENTRIES_COUNT:
            raise serializers.ValidationError(
                f'Analytical entires count must be less than {settings.ANALYTICAL_ENTRIES_COUNT}'
             )
```